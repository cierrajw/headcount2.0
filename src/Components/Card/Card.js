import React from 'react';

const Card = ({location, stats}) =>{
  const statsArray = Object.keys(stats);
  const districtData = statsArray.map(data=>stats[data])
  return(
    <div>
    <h4>{location}</h4>
    <h3>{districtData}</h3>
    </div>
  )
}

export default Card;
