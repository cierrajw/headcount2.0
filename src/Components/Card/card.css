.card-div{
  border: 1px solid blue;
  width: 14em;
  height: 30em;
  margin: 0 auto;
  word-wrap: break-word;
  text-align: center;
  padding: 1em;
  margin-bottom: 2em;

}
