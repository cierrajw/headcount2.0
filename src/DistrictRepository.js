

class DistrictRepository{
  constructor(stats){
    super();
    this.stats = {}
  }

  // findByName()
  //
  // findAllMatching()
}

export default DistrictRepository;
