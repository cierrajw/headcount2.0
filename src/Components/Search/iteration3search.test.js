import React from 'react';
import { mount, shallow } from 'enzyme';
import ReactDOM from 'react-dom';
import Card from './Card'
import DistrictRepository from '../../helper.js';
import kinderData from '../../data/kindergartners_in_full_day_program.js';

describe('Search', () =>  {

  const mockData = {
      district: 'ADAMS'
    }

  const district = new DistrictRepository(kinderData);

  it('should match the snapshot', () => {

    const wrapper = shallow(<Search {...mockData}/>);

    expect(wrapper).toMatchSnapshot();
  });

  it('should c', () => {

    const wrapper = shallow(<Search {...mockData}/>);

    const expectedState = { district: 'Adams'}



    expect(wrapper).toMatchSnapshot();
  });

});
