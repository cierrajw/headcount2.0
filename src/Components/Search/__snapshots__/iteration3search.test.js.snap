// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search
    district="ADAMS"
    searchData={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        className="submit-form"
      >
        <input
          className="search-input"
          name="district"
          onChange={[Function]}
          placeholder="Search Districts"
        />
      </form>,
      "className": "search-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <input
          className="search-input"
          name="district"
          onChange={[Function]}
          placeholder="Search Districts"
        />,
        "className": "submit-form",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "search-input",
          "name": "district",
          "onChange": [Function],
          "placeholder": "Search Districts",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          className="submit-form"
        >
          <input
            className="search-input"
            name="district"
            onChange={[Function]}
            placeholder="Search Districts"
          />
        </form>,
        "className": "search-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <input
            className="search-input"
            name="district"
            onChange={[Function]}
            placeholder="Search Districts"
          />,
          "className": "submit-form",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "search-input",
            "name": "district",
            "onChange": [Function],
            "placeholder": "Search Districts",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
